### PR TITLE
Add v1 users schema and controller stubs

### PR DESCRIPTION
**Pull request by keithmanville**
Replacing #438 with PR for branch on `usnistgov/dioptra` instead of fork `pscemama-mitre/dioptra`

### CODE REVIEW

**Comment by keithmanville** [comment]
This looks good to me. Recommend squashing commits and merging.

File: src/dioptra/restapi/v1/users/schema.py [hunk 1]
```diff
@@ -37,3 +44,102 @@ class UserRefSchema(Schema):
...
+        attribute="id",
+        metadata=dict(description="ID for the User resource."),
+        dump_only=True,
+    )
+
+
+class UserCurrentSchema(UserSchema):
+    """The schema for the currently logged-in User."""
+
+    memberOf = fields.Nested(
```
Comment (jkglasbrenner) [inline src/dioptra/restapi/v1/users/schema.py:80]: Let's change this field to `groups` (also `attribute="groups"`).

File: src/dioptra/restapi/v1/users/schema.py [hunk 2]
```diff
@@ -37,3 +44,102 @@ class UserRefSchema(Schema):
...
+
+class UserSchema(UserMutableFieldsSchema):
+    """The schema for a User that is not the logged-in User."""
+
+    password = fields.String(
+        attribute="password",
+        metadata=dict(description="Password for the User resource."),
+        load_only=True,
+    )
+
```
Comment (jkglasbrenner) [inline src/dioptra/restapi/v1/users/schema.py:69]: Also needs a `confirmPassword` field for initial user registration, so its `load_only=True`.

https://github.com/usnistgov/dioptra/blob/60af0a6aac3af7e4e9a47e16afa0e6877af9f05f/src/dioptra/restapi/v0/user/schema.py#L49-L55

File: src/dioptra/restapi/v1/users/controller.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        log = LOGGER.new(
+            request_id=str(uuid.uuid4()),
+            resource="User",
+            request_type="DELETE",
+        )
+        log.debug("Request received")
+
+    @login_required
+    @accepts(schema=UserMutableFieldsSchema, api=api)
+    @responds(schema=UserSchema, api=api)
```
Comment (jkglasbrenner) [inline src/dioptra/restapi/v1/users/controller.py:110]: The current PUT should return the same fields that a /current GET provides.

```suggestion
    @responds(schema=UserCurrentSchema, api=api)
```

File: src/dioptra/restapi/v1/users/schema.py [hunk 4]
```diff
@@ -37,3 +44,102 @@ class UserRefSchema(Schema):
...
+    oldPassword = fields.String(
+        attribute="old_password",
+        metadata=dict(description="Old password for the User resource."),
+    )
+
+    newPassword = fields.String(
+        attribute="new_password",
+        metadata=dict(description="New password for the User resource."),
+    )
+
```
Comment (jkglasbrenner) [inline src/dioptra/restapi/v1/users/schema.py:129]: Also needs a `confirmNewPassword` field for initial user registration, so its `load_only=True`.

https://github.com/usnistgov/dioptra/blob/60af0a6aac3af7e4e9a47e16afa0e6877af9f05f/src/dioptra/restapi/v0/user/schema.py#L49-L55

**Review by jkglasbrenner — CHANGES_REQUESTED** [review]

**Comment by pscemama-mitre** [comment]
Getting this output when trying to run

```bash
python3 -m tox run -e black,isort,flake8,mypy,gitlint -- src/dioptra
```

```bash
src/dioptra/restapi/v1/users/schema.py:141:90: B950 line too long (89 > 80 characters)
src/dioptra/rq/tasks/run_mlflow.py:206:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:262:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:265:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:292:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:293:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:294:21: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:337:25: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/sdk/object_detection/bounding_boxes/postprocessing/tensorflow_backend/confluence.py:354:17: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/task_engine/run_experiment.py:148:13: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
src/dioptra/task_engine/util.py:317:9: B909 editing a loop's mutable iterable often leads to unexpected results/bugs
```

@keithmanville  @jkglasbrenner I believe I'm doing something wrong?

**Comment by pscemama-mitre** [comment]
@keithmanville this is the question I had (also cc @jkglasbrenner)

Right now it seems to me like we allow a User's password to be changed via ID (and knowing the old password).

We have both `/current/password` and `id/password`. Following this, why do we not allow a User's email/username to be changed via ID? 

Or, why do we even bother having a User's password being able to be changed via ID (and not current?)?

**Comment by keithmanville** [comment]
I believe the intent behind `/users/{id}/password` is for a user to change their password without being logged in if their password has expired, or they are locked out for some other reason. Other updates to their profile (like username or email changes) should happen once logged in and they are now the current user.

**Pull request merged**